instrumentation/aws/xray/AwsXrayIdGenerator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace Instrumentation\Aws\Xray;
+
+use OpenTelemetry\Sdk\Trace\RandomIdGenerator;
+use OpenTelemetry\Sdk\Trace\IdGenerator;
+
+/**
+ * An ID generator that generates trace IDs that conforms to AWS X-Ray format
+ * Refer to the AWS X-Ray documentation: 
+ * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids
+ */
+class AwsXrayIdGenerator implements IdGenerator
+{
+    private const TRACE_ID_RANDOM_HEX_LENGTH = 24;
+    private const SPAN_ID_HEX_LENGTH = 16;
+
+    /**
+     * Returns a trace ID in AWS X-Ray format. 
+     * Returns a version number, an epoch timestamp converted to 
+     * 8 hexadecimal digits, and random 96-bit string converted to a
+     * 24 character hexadecimal string, all seperated by 2 dashes.
+     * 
+     * Example: 1-60cd399d-76b435b5411c66f29b238014
+     */
+    public function generateTraceId(): string
+    {   
+        $idGenerator = new RandomIdGenerator();
+        return dechex(time()) . $idGenerator->randomHex(self::TRACE_ID_RANDOM_HEX_LENGTH);
+    }
+
+    /**
+     * Returns a random 64-bit string in the form of 16 
+     * hexadecimal characters. 
+     */
+    public function generateSpanId(): string
+    {
+        $idGenerator = new RandomIdGenerator;
```

Comment:
Why is there no bracket when initializing `RandomIdGenerator` compared to above? Maybe both are good in php, but better keep the same format.

Reply:
Oops, that was a typo, fixed!